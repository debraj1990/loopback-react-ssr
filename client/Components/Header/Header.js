import React from 'react'

const Header = () => {

    return (
        <nav>
            <div className="nav-wrapper">
                <Link to="/" className="brand-logo">
                React SSR
                </Link>
                <ul className="right">
                <li>
                    <Link to="/users">Users</Link>
                </li>
                <li>
                    <Link to="/admins">Admins</Link>
                </li>
                <li>{authButton}</li>
                </ul>
            </div>
        </nav>
    )
}

