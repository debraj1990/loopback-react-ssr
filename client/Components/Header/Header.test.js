import React from 'react';
import {configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';

import { Header } from './Header';


configure({adapter: new Adapter });


describe('<Header />', () => {
    

    it('should render the Header Component', () => {
        const component = shallow(<Header />)
        const tree = toJson(component)
        expect(tree).toMatchSnapshot()
    })

    it('should render Header with login link if auth prop is false', () => {
        const component = shallow(<Header auth={false} />)
        expect(component.contains( <a href="/api/auth/google">Login</a>)).toBeTruthy()
    })

    it('should render Header with logout link if auth prop is true', () => {
        const component = shallow(<Header auth />)
        expect(component.contains(<a href="/api/logout">Logout</a>)).toBeTruthy()
    })
})