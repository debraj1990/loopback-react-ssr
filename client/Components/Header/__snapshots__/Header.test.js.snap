// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the Header Component 1`] = `
<nav>
  <div
    className="nav-wrapper"
  >
    <Link
      className="brand-logo"
      replace={false}
      to="/"
    >
      SkillNav
    </Link>
    <ul
      className="right"
    >
      <li>
        <Link
          replace={false}
          to="/users"
        >
          Users
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/admins"
        >
          Admins
        </Link>
      </li>
      <li>
        <a
          href="/api/auth/google"
        >
          Login
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
